src/commands/code-analyzer/run.ts

```diff
@@ -67,8 +75,58 @@ export default class RunCommand extends SfCommand<void> {
 
 	public async run(): Promise<void> {
 		const parsedFlags = (await this.parse(RunCommand)).flags;
-		const action: RunAction = new RunAction();
-		action.execute(parsedFlags);
+		const dependencies: RunDependencies = this.createDependencies(parsedFlags.view as View, parsedFlags['output-file']);
+		const action: RunAction = RunAction.createAction(dependencies);
+		const runInput: RunInput = {
+			'config-file': parsedFlags['config-file'],
+			'path-start': parsedFlags['path-start'],
+			'rule-selector': parsedFlags['rule-selector'],
+			'workspace': parsedFlags['workspace']
+		};
+		const possibleThreshold = this.convertThresholdToEnum(parsedFlags['severity-threshold']);
+		if (possibleThreshold) {
+			runInput['severity-threshold'] = possibleThreshold;
+		}
```

Comment:
I actually tried that, but then the `--severity-threshold > is unused if not directly specified` test at line 187 of `run.test.ts` started failing the `.not.toHaveProperty` assertion. Apparently, Jest sees some kind of difference between an absent property and one that's present as `undefined`.

Reply:
Yes there is a difference for sure... but only with regards to the way you defined your assertion. Instead of .not.toHaveProperty consider using .toBeUndefined.